src/update_roster.R

```diff
@@ -1,97 +1,276 @@
+        dplyr::select(-c(status, height, weight, headshot)) |>
         dplyr::left_join(
           shield |>
-            dplyr::select(gsis_id, birth_date, height) |>
-            dplyr::distinct(),
+            dplyr::select(gsis_id, status, height, weight, headshot = headshot_url, birth_date) |>
+            dplyr::arrange(gsis_id, status) |>
+            dplyr::distinct(gsis_id, .keep_all = TRUE),
           by = c("gsis_id"),
           na_matches = "never"
         ) |>
         fill_ids()
     }
-  }
 
-  if (season >= 2016 && nrow(weekly_rosters) == 0) {
-    roster <- shield
-  }
+    if (season >= 2016) {
+      weekly_rosters <- build_rosters_weekly_ngsapi(season)
+      if (nrow(weekly_rosters) > 0) {
+        roster <- convert_weekly_to_season_rosters(weekly_rosters)
+        shield <- build_rosters_shieldapi(season)
+        weekly_rosters <- weekly_rosters |>
+          dplyr::select(-c(height)) |>
+          dplyr::left_join(
+            shield |>
+              dplyr::select(gsis_id, birth_date, height) |>
+              dplyr::distinct(),
+            by = c("gsis_id"),
+            na_matches = "never"
+          ) |>
+          fill_ids()
+
+        roster <- roster |>
+          dplyr::select(-c(height)) |>
+          dplyr::left_join(
+            shield |>
+              dplyr::select(gsis_id, birth_date, height) |>
+              dplyr::distinct(),
+            by = c("gsis_id"),
+            na_matches = "never"
+          ) |>
+          fill_ids()
+      }
+    }
 
-  if ((season < 2002 && length(roster) == 0) | (season >= 2002 && length(weekly_rosters) == 0)) {
-    return(invisible(NULL))
-  }
+    if (season >= 2016 && nrow(weekly_rosters) == 0) {
+      roster <- shield
+    }
+
+    if ((season < 2002 &&
+         length(roster) == 0) |
+        (season >= 2002 && length(weekly_rosters) == 0)) {
+      return(invisible(NULL))
+    }
+
+    if (season >= 2002) {
+      weekly_rosters <- weekly_rosters |>
+        dplyr::mutate(
+          dplyr::across(
+            c(gsis_it_id,
+              smart_id),
+            as.character
+          ),
+          dplyr::across(
+            c(entry_year,
+              rookie_year),
+            as.integer
+          )
+        ) |>
+        dplyr::left_join(
+          nflreadr::load_players() |>
+            dplyr::select(
+              gsis_id,
+              full_name = display_name,
+              first_name,
+              last_name,
+              esb_id,
+              birth_date,
+              college = college_name,
+              entry_year,
+              rookie_year,
+              draft_club,
+              gsis_it_id,
+              smart_id,
+              height,
+              weight
+            ) |>
+            dplyr::mutate(
+              birth_date = as.Date(birth_date),
+              dplyr::across(
+                c(gsis_it_id,
+                  smart_id),
+                as.character
+              )
+            ),
```

Comment:
Looks like this block of logic is repeated again in L190, should this be calculated into a df_players around L4 instead?